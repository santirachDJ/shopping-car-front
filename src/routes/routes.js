import home from "../pages/home/home";
import ShoppingCar from "../pages/shoppingCar/shoppingCar";
import DetailShoppingCar from "../pages/detailShoppingCar/detailShoppingCar"
const routes = [
  {
    path: "/",
    component: ShoppingCar,
    exact: true,
    name: "Crear shopping",
  },
  {
    path: "/shopping/product",
    component: home,
    exact: true,
    name: "Ver productos",
  },
  {
    path: "/shopping/details",
    component: DetailShoppingCar,
    exact: true,
    name: "Ver shopping car",
  },
  
  
];

export default routes;
