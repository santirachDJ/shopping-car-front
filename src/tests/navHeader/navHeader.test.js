import React from 'react'
import { mount } from 'enzyme'
import Nav from "emerald-ui/lib/Nav";
import NavHeader from '../../components/layout/navHeader.layout'
import { findElement } from '../utils/utilsEnzyme'
import { act } from 'react-dom/test-utils';
import routes from '../../routes/routes';
import { MemoryRouter } from 'react-router-dom';
/** aplicandole test a un componente que requeire de un useContext */
const mockHistoryPush = jest.fn();
const mockHistoryGoBack = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: jest.fn().mockReturnValue({
      pathname: '/another-route',
      search: '?shopping=123456',
      hash: '',
      state: null,
      key: '5nvxpbdafa',
    }),
    useHistory: () => ({
        push: mockHistoryPush,
        goBack: mockHistoryGoBack,
    }),
}));
const waitForComponentToPaint = async (wrapper) => {
    await act(async () => {
        await new Promise((resolve) => setTimeout(resolve, 0));
        wrapper.update();
    });
};
describe('navHeader', () => {
    let wrapper


    beforeEach(() => {
        wrapper = mount(
            <NavHeader/>
        );
        waitForComponentToPaint(wrapper); 
    });

    afterEach(() => {
        wrapper = {};
    });

    it('Matches the snapshot', () => {
        expect(wrapper.getElements()).toMatchSnapshot()
    })

    it('Find element componenet', () => {
        expect(findElement(wrapper, 'div').exists()).toBe(true)
    })

    test('Should have options', async () => {
        const table = wrapper.find(Nav);
        expect(table.find('a').length).toEqual(routes.length);
    });

    test('triggers path change', () => {
      
        const element = findElement(wrapper,'a');
        expect(element.exists()).toBeTruthy();
        element.simulate("click");
        expect(mockHistoryPush).toBeCalled();
        expect(mockHistoryPush).toBeCalledWith({"pathname": "/", "search": "?shopping=123456", "state": {"shoppingId": "123456"}});
      });

})

