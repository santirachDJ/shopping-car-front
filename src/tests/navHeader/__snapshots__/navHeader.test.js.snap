// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navHeader Matches the snapshot 1`] = `
Array [
  <NavHeader />,
]
`;
